app/controllers/courses_controller.rb

```diff
@@ -11,6 +11,16 @@ def show
   end
 
   def new
+    @course = Course.new(chef_params)
+    @course.user = current_user
```

Comment:
This @course.user here might be a wrong value assign because the course instance doesn't have a user instance variable (I tested in rails console and I can't access to course.user, correct me if I'm wrong). I guess this is the correct way of what you're trying to do `@course.chef_profile = current_user.chef_profile`.